### PR TITLE
Add state object

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Review by Meg-Sowka — APPROVED** [review]
It's perfect! Good job! :)

File: src/transformState.js [hunk 1]
```diff
@@ -5,7 +5,32 @@
...
+        }
+        break;
+      }
+
+      default: {
+        throw new Error("Incorrect action type");
+      }
+    }
+  }
 }
```
Comment (choeqq) [inline src/transformState.js]: function should return state

File: src/transformState.js [hunk 2]
```diff
@@ -5,7 +5,32 @@
...
+    switch (action.type) {
+      case "addProperties": {
+        Object.assign(state, action.extraData);
+        break;
+      }
+
+      case "removeProperties": {
+        for (const key of action.keysToRemove) {
+          delete state[key];
+        }
```
Comment (choeqq) [inline src/transformState.js:18]: it would be nice to check if property we want to delete exists

**Review by choeqq — CHANGES_REQUESTED** [review]

**Review by choeqq — APPROVED** [review]
Good job 🚀

**Pull request closed without merging**